packages/cli/src/cli/basemaps/action.export.ts

```diff
@@ -0,0 +1,141 @@
+    if (item.rules == null) {
+        table.TileSet.migrate(item as any); // migrate old format
+    }
+
+    const imageryMap = await primeImageryCache(new Set(item.rules.map((r) => r.imgId)));
+
+    const imagery = item.rules
+        .map(
+            (rule): FullImageryConfig => {
+                const image = imageryMap.get(rule.imgId);
+                if (image == null) {
+                    throw new Error(`Can't find imagery record "${rule.imgId}" for Tileset "${item.id}"`);
+                }
+                return {
+                    id: TileMetadataTable.unprefix(RecordPrefix.Imagery, rule.imgId),
+                    name: image.name,
+                    priority: rule.priority,
+                    minZoom: rule.minZoom,
+                    maxZoom: rule.maxZoom,
+                };
+            },
+        )
+        .sort(compareNamePriority)
+        .map((r) => removeDefaults(defaults, r));
+
+    return {
+        name,
+        projection: projection.code,
+        background: rgbaToHex(item.background ?? DefaultBackground),
+        defaults,
+        imagery,
+    };
+}
+
+export class ExportAction extends TileSetBaseAction {
+    private output: CommandLineStringParameter;
+    private defaults: CommandLineStringParameter;
+    private full: CommandLineFlagParameter;
+    private tag: CommandLineStringParameter;
+
+    public constructor() {
+        super({
+            actionName: 'export',
+            summary: 'Export imagery and tileset rules',
+            documentation: '',
+        });
+    }
+
+    protected onDefineParameters(): void {
+        super.onDefineParameters();
+        this.output = this.defineStringParameter({
+            argumentName: 'JSON_FILE',
+            parameterLongName: '--output',
+            parameterShortName: '-o',
+            description: 'File to output to',
+            required: true,
+        });
+        this.defaults = this.defineStringParameter({
+            argumentName: 'JSON_FILE',
+            parameterLongName: '--defaults',
+            description:
+                'Config File containing rule defaults. This is added to the output and used to minimize the rule fields. May not be used with --full',
+            required: false,
+        });
+        this.full = this.defineFlagParameter({
+            parameterLongName: '--full',
+            description: 'Export the entire tilemetadatdb',
+            required: false,
+        });
+        defineTagParameter(this);
+    }
+
+    protected async onExecute(): Promise<void> {
+        const outFile = this.output.value!;
+        const table = new TileMetadataTable();
+
+        let output: string;
+        if (this.full.value) {
+            const lines: string[] = [];
+            for await (const item of table) {
+                lines.push(JSON.stringify(item));
+            }
+            output = `[\n  ${lines.join(',\n  ')}\n]`;
```

Comment:
why are you string bashing JSON? cant you just `JSON.stringify(lines)` if you remove the internal stringify?

Reply:
Otherwise just one long line but don't want to prettify the whole thing as then too many lines. All is mute because I am removing the `full` option